cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/ServiceImpl.kt

```diff
@@ -367,50 +366,52 @@ private class ServiceImpl(
         consentActionImpl: ConsentActionImpl,
         env: Env,
         sPConsentsSuccess: ((SPConsents) -> Unit)?,
-        pmId: String?
+        pmId: String? // TODO figure out if we actually need this param, since we DO NOT use it
     ): Either<ChoiceResp> {
         return when (consentActionImpl.campaignType) {
             GDPR -> {
                 sendConsentGdpr(
-                    consentActionImpl,
-                    env,
-                    sPConsentsSuccess
-                ).map { ChoiceResp(gdpr = it) }
+                    env = env,
+                    consentAction = consentActionImpl,
+                    onSpConsentsSuccess = sPConsentsSuccess,
+                ).map { gdpr -> ChoiceResp(gdpr = gdpr) }
             }
             CCPA -> {
                 sendConsentCcpa(
-                    consentActionImpl,
-                    env,
-                    sPConsentsSuccess
-                ).map { ChoiceResp(ccpa = it) }
+                    env = env,
+                    consentAction = consentActionImpl,
+                    onSpConsentsSuccess = sPConsentsSuccess,
+                ).map { ccpa -> ChoiceResp(ccpa = ccpa) }
             }
             USNAT -> {
                 sendConsentUsNat(
                     env = env,
                     consentAction = consentActionImpl,
-                    onConsentSuccess = sPConsentsSuccess,
-                ).map { ChoiceResp(usNat = it) }
+                    onSpConsentSuccess = sPConsentsSuccess,
+                ).map { usNat -> ChoiceResp(usNat = usNat) }
             }
         }
     }
 
     fun sendConsentGdpr(
-        consentActionImpl: ConsentActionImpl,
         env: Env,
-        sPConsentsSuccess: ((SPConsents) -> Unit)?
+        consentAction: ConsentActionImpl,
+        onSpConsentsSuccess: ((SPConsents) -> Unit)?
     ): Either<GdprCS> = check {
 
         var getResp: ChoiceResp? = null
 
-        val actionType = consentActionImpl.actionType
-        if (actionType == ActionType.ACCEPT_ALL || actionType == ActionType.REJECT_ALL) {
+        if (consentAction.actionType.isAcceptOrRejectAll()) {
 
             val getChoiceParamReq = GetChoiceParamReq(
-                choiceType = consentActionImpl.actionType.toChoiceTypeParam(),
+                choiceType = consentAction.actionType.toChoiceTypeParam(),
                 accountId = spConfig.accountId.toLong(),
                 propertyId = spConfig.propertyId.toLong(),
                 env = env,
-                metadataArg = campaignManager.metaDataResp?.toMetaDataArg()?.copy(ccpa = null),
+                metadataArg = campaignManager.metaDataResp?.toMetaDataArg()?.copy(
+                    ccpa = null,
+                    usNat = null,
+                ),
```

Comment:
@carmelo-iriti in order to get choice for `GDPR` only, we have to null out meta data for other campaigns, i.e. `CCPA` and `UsNat`

Reply:
let me take a look I don’t remember this detail